packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -530,6 +531,14 @@ export function TransferPanel() {
         return undefined
       }
 
+      if (
+        isDepositMode &&
+        l2NetworkChainId &&
+        selectedToken &&
+        isWithdrawOnlyToken(selectedToken.address, l2NetworkChainId)) {
+          return TransferPanelMainErrorMessage.WITHDRAW_ONLY
+      }
```

Comment:
We're slowly gonna deprecate the `useAppState` hook, so you can fetch the `l2Network` from `useNetworksAndSigners`:
```
if (isDepositMode && selectedToken && isWithdrawOnlyToken(selectedToken.address, l2Network.chainID)) {
  return TransferPanelMainErrorMessage.WITHDRAW_ONLY
}
```